modules/test/playwright/tests/style-book-web/previewStyleBookOnDisplayPageTemplates.spec.ts

```diff
@@ -0,0 +1,324 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page, expect, mergeTests} from '@playwright/test';
+
+import {loginTest} from '../../fixtures/loginTest';
+import {pageEditorPagesTest} from '../../fixtures/pageEditorPagesTest';
+import {styleBookPageTest} from '../../fixtures/styleBookPageTest';
+import {workflowPagesTest} from '../../fixtures/workflowPagesTest';
+import {PageEditorPage} from '../../pages/layout-content-page-editor-web/PageEditorPage';
+import {DisplayPageTemplatesPage} from '../../pages/layout-page-template-admin-web/DisplayPageTemplatesPage';
+import {StyleBooksPage} from '../../pages/style-book-web/StyleBooksPage';
+import fillAndClickOutside from '../../utils/fillAndClickOutside';
+import getRandomString from '../../utils/getRandomString';
+import {waitForSuccessAlert} from '../../utils/waitForSuccessAlert';
+import {blogsPagesTest} from '../blogs-web/fixtures/blogsPagesTest';
+
+const test = mergeTests(
+	blogsPagesTest,
+	loginTest(),
+	pageEditorPagesTest,
+	styleBookPageTest,
+	workflowPagesTest
+);
+
+const addHeading = async (
+	pageEditorPage: PageEditorPage,
+	publishButton: Locator
+) => {
+	const headingId = await pageEditorPage.getFragmentId('Heading');
+
+	expect(await pageEditorPage.isActive(headingId)).toBe(true);
+
+	await publishButton.click();
+};
+
+const changeStyleAndCheckPreview = async (
+	colorHEX: string,
+	colorRGB: string,
+	displayPageTemplatesPage: DisplayPageTemplatesPage,
+	page: Page,
+	pageEditorPage: PageEditorPage,
+	styleBooksPage: StyleBooksPage,
+	templateName: string
+) => {
+	const publishButton = page.getByRole('button', {
+		name: 'Publish',
+	});
+
+	await styleBooksPage.goto();
+	await page.getByRole('button', {name: /more actions/i}).click();
+	await page.getByRole('menuitem', {name: 'Edit'}).click();
```

Comment:
Editing a style book should be handled by the styleBooksPage. This should be moved there and it should also account for editing a specific style book when more than one style book exists. If it doesn't handle that it will be very fragile and break easily.